centreon/src/Core/Security/ProviderConfiguration/Domain/SecurityAccess/AttributePath/AttributePathFetcherInterface.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  For more information : contact@centreon.com
+ */
+
+declare(strict_types=1);
+
+namespace Core\Security\ProviderConfiguration\Domain\SecurityAccess\AttributePath;
+
+use Core\Security\Authentication\Domain\Model\ProviderToken;
+use Core\Security\ProviderConfiguration\Domain\Model\Configuration;
+use Core\Security\ProviderConfiguration\Domain\Model\Endpoint;
+
+interface AttributePathFetcherInterface
+{
+    /**
+     * @param string $token
+     * @param Configuration $configuration
+     * @param Endpoint $endpoint
+     * @return array
+     */
+    public function fetch(string $token, Configuration $configuration, Endpoint $endpoint): array;
+
+    /**
+     * @param $data
+     * @return bool
+     */
+    public function supports($data): bool;
```

Comment:
```suggestion
    public function supports($data): bool;
```
missing  type hint on $data

Reply:
Or explicit mixed !
```suggestion
    public function supports(mixed $data): bool;
```

Reply:
With this pattern you can pass anything to perform your check, it can be an Endpoint object, a string, an array... a mixed is wille be appropriate, but indeed in this case we can pass an endpoint object